src/core/thread/dua_manager.cpp

```diff
@@ -483,15 +483,18 @@ void DuaManager::PerformNextRegistration(void)
 
             if (mChildDuaMask.Get(childIndex) && !mChildDuaRegisteredMask.Get(childIndex))
             {
-                mChildIndexDuaRegistering = childIndex;
-                break;
+                child  = Get<ChildTable>().GetChildAtIndex(childIndex);
```

Comment:
I think parent should not set `mChildDuaMask` if the Child's DUA is not successfully added. 

We may need to fix `MleRouter::UpdateChildAddresses` to call `UpdateChildDomainUnicastAddress` correctly. 

Thoughts? @librasungirl @canisLupus1313

Reply:
I agree. I submitted an alternative proposal in #7164. Please review. Thanks.

Reply:
@jwhui @simonlingoogle 
#7164 looks good for me.
One more question what is there a possibility that for loop from line 480 will end without finding any index?

Reply:
> One more question what is there a possibility that for loop from line 480 will end without finding any index?

In theory it should not happen  because we have checked `(mChildDuaMask.HasAny() && mChildDuaMask != mChildDuaRegisteredMask)` at above. 

If we can't fiind a Child index to register DUA, it should generally mean we are not updating `mChildDuaMask` correctly, which is a bug that is proper to assert.